android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/ContentStateReducer.kt

```diff
@@ -299,6 +299,14 @@ internal object ContentStateReducer {
             is ContentAction.UpdateExpandedToolbarStateAction -> updateContentState(state, action.sessionId) {
                 it.copy(showToolbarAsExpanded = action.expanded)
             }
+            is ContentAction.CheckForFormDataAction -> updateContentState(state, action.tabId) {
+                // todocathy there is no store parameter to update, but this cannot be removed
+                it.copy(desktopMode = action.containsFormData)
+            }
+            is ContentAction.ClearHighPrioritySessionAction -> updateContentState(state, action.tabId) {
+                // todocathy there is no store parameter to update, but this cannot be removed
+                it.copy(desktopMode = action.tabId == "0")
+            }
```

Comment:
Should we create a new property on [ContentState](https://github.com/mozilla-mobile/firefox-android/blob/e1ccf1865c6f8485801c05a544d80f52032006a4/android-components/components/browser/state/src/main/java/mozilla/components/browser/state/state/ContentState.kt#L64) to represent `containsFormData`?

Reply:
I'm working on updating this to throw an exception that should direct users to use the middleware instead. 
`throw IllegalStateException("You need to add SessionPrioritizationMiddleware to your BrowserStore. ($action)")`